ambry-api/src/test/java/com.github.ambry/store/MockWrite.java

```diff
@@ -40,14 +40,20 @@ public int appendFrom(ByteBuffer buffer) {
   }
 
   @Override
-  public void appendFrom(ReadableByteChannel channel, long size) throws IOException {
+  public void appendFrom(ReadableByteChannel channel, long size) throws StoreException {
     if (size > buf.remaining()) {
       throw new IllegalStateException("Cannot fit " + size + " bytes in buf that has " + buf.remaining() + " space");
     }
     int savedLimit = buf.limit();
     buf.limit(buf.position() + (int) size);
-    while (buf.hasRemaining()) {
-      channel.read(buf);
+    try {
+      while (buf.hasRemaining()) {
+        channel.read(buf);
+      }
+    } catch (IOException e) {
+      StoreErrorCodes errorCode =
+          e.getMessage().equals(StoreException.IO_ERROR_STR) ? StoreErrorCodes.IOError : StoreErrorCodes.Unknown_Error;
```

Comment:
this lookup is a bit imprecise as the actual text might change with later java versions, but I understand that there is nothing better at this time. Just perhaps add a comment here about it?